framework/src/predictors/AdamsPredictor.C

```diff
@@ -73,6 +73,10 @@ AdamsPredictor::apply(NumericVector<Number> & sln)
   // that gets called on time step begin.
   // That means that history control must go here.
   historyControl();
+
+  if (!shouldApply())
```

Comment:
It looks like our wires got crossed.  So is the name ```initialize()``` preferred to ```timestepSetup()```? Would it get called at the beginning of the timestep or when the predictor is about to be applied?  Either way works.